internal/pkg/cli/pipeline_deploy.go

```diff
@@ -180,17 +189,30 @@ func (o *deployPipelineOpts) Execute() error {
 }
 
 func (o *deployPipelineOpts) validatePipelineName() error {
-	data, err := o.ws.ReadPipelineManifest()
+	pipelines, err := o.ws.ListPipelines()
 	if err != nil {
-		return fmt.Errorf("read pipeline manifest: %w", err)
+		return fmt.Errorf("list pipelines: %w", err)
 	}
-	pipeline, err := manifest.UnmarshalPipeline(data)
-	if err != nil {
-		return fmt.Errorf("unmarshal pipeline manifest: %w", err)
+	for name, path := range pipelines {
+		if name == o.name {
+			o.path = path
```

Comment:
It totally makes sense to put `o.path = path` here for performance reasons, but I feel like the most robust place where we need to guarantee that `path` has been set is in `Execute`.

What do you think of additionally creating a method for `Execute` that's like:
```go
func (o *deployPipelineOpts) Execute() error {
   if err := cachePipelinePath(); err != nil { return nil }
   ...
}

func (o *deployPipelineOpts) cachePipelinePath() error {
   if o.path != "" { return nil }
   pipelines, err := o.ws.ListPipelines()
   ...
}
```